app/src/main/java/com/hedvig/app/feature/keygear/KeyGearValuationActivity.kt

```diff
@@ -28,16 +44,51 @@ class KeyGearValuationActivity : BaseActivity(R.layout.activity_key_gear_valuati
         }
 
         continueButton.setHapticClickListener {
-            //TODO
-            model.submit()
-            onBackPressed()
+            val price = priceInput.getText()
+            date?.let { date ->
```

Comment:
When you need to null-check multiple values, you can use a neat little function called `safeLet`.

You provide `safeLet` with `n` nullable values, and it only executes the provided block of code when all of them are non-null.

Example:

```kotlin
import com.hedvig.app.util.safeLet
// ...
val a: String? = null
val b = 123
safeLet(a, b) { a, b ->
    // This block does not execute because `a` is null
    print(a)
    print(b)
}

val c = false
val d = Pair(1, 2)
safeLet(c, d) { c, d ->
    // This block of code does execute, because c and d are both non-null
    print(c)
    print(d)
}
```